components/tidb_query/src/expr_util/collation/utf8mb4.rs

```diff
@@ -0,0 +1,613 @@
+    [0x01A0, 0x01A1, 0x004F],
+    [0x01A2, 0x01A3, 0x01A2],
+    [0x01A2, 0x01A3, 0x01A2],
+    [0x01A4, 0x01A5, 0x01A4],
+    [0x01A4, 0x01A5, 0x01A4],
+    [0x01A6, 0x0280, 0x01A6],
+    [0x01A7, 0x01A8, 0x01A7],
+    [0x01A7, 0x01A8, 0x01A7],
+    [0x01A9, 0x0283, 0x01A9],
+    [0x01AA, 0x01AA, 0x01AA],
+    [0x01AB, 0x01AB, 0x01AB],
+    [0x01AC, 0x01AD, 0x01AC],
+    [0x01AC, 0x01AD, 0x01AC],
+    [0x01AE, 0x0288, 0x01AE],
+    [0x01AF, 0x01B0, 0x0055],
+    [0x01AF, 0x01B0, 0x0055],
+    [0x01B1, 0x028A, 0x01B1],
+    [0x01B2, 0x028B, 0x01B2],
+    [0x01B3, 0x01B4, 0x01B3],
+    [0x01B3, 0x01B4, 0x01B3],
+    [0x01B5, 0x01B6, 0x01B5],
+    [0x01B5, 0x01B6, 0x01B5],
+    [0x01B7, 0x0292, 0x01B7],
+    [0x01B8, 0x01B9, 0x01B8],
+    [0x01B8, 0x01B9, 0x01B8],
+    [0x01BA, 0x01BA, 0x01BA],
+    [0x01BB, 0x01BB, 0x01BB],
+    [0x01BC, 0x01BD, 0x01BC],
+    [0x01BC, 0x01BD, 0x01BC],
+    [0x01BE, 0x01BE, 0x01BE],
+    [0x01F7, 0x01BF, 0x01F7],
+    [0x01C0, 0x01C0, 0x01C0],
+    [0x01C1, 0x01C1, 0x01C1],
+    [0x01C2, 0x01C2, 0x01C2],
+    [0x01C3, 0x01C3, 0x01C3],
+    [0x01C4, 0x01C6, 0x01C4],
+    [0x01C4, 0x01C6, 0x01C4],
+    [0x01C4, 0x01C6, 0x01C4],
+    [0x01C7, 0x01C9, 0x01C7],
+    [0x01C7, 0x01C9, 0x01C7],
+    [0x01C7, 0x01C9, 0x01C7],
+    [0x01CA, 0x01CC, 0x01CA],
+    [0x01CA, 0x01CC, 0x01CA],
+    [0x01CA, 0x01CC, 0x01CA],
+    [0x01CD, 0x01CE, 0x0041],
+    [0x01CD, 0x01CE, 0x0041],
+    [0x01CF, 0x01D0, 0x0049],
+    [0x01CF, 0x01D0, 0x0049],
+    [0x01D1, 0x01D2, 0x004F],
+    [0x01D1, 0x01D2, 0x004F],
+    [0x01D3, 0x01D4, 0x0055],
+    [0x01D3, 0x01D4, 0x0055],
+    [0x01D5, 0x01D6, 0x0055],
+    [0x01D5, 0x01D6, 0x0055],
+    [0x01D7, 0x01D8, 0x0055],
+    [0x01D7, 0x01D8, 0x0055],
+    [0x01D9, 0x01DA, 0x0055],
+    [0x01D9, 0x01DA, 0x0055],
+    [0x01DB, 0x01DC, 0x0055],
+    [0x01DB, 0x01DC, 0x0055],
+    [0x018E, 0x01DD, 0x018E],
+    [0x01DE, 0x01DF, 0x0041],
+    [0x01DE, 0x01DF, 0x0041],
+    [0x01E0, 0x01E1, 0x0041],
+    [0x01E0, 0x01E1, 0x0041],
+    [0x01E2, 0x01E3, 0x00C6],
+    [0x01E2, 0x01E3, 0x00C6],
+    [0x01E4, 0x01E5, 0x01E4],
+    [0x01E4, 0x01E5, 0x01E4],
+    [0x01E6, 0x01E7, 0x0047],
+    [0x01E6, 0x01E7, 0x0047],
+    [0x01E8, 0x01E9, 0x004B],
+    [0x01E8, 0x01E9, 0x004B],
+    [0x01EA, 0x01EB, 0x004F],
+    [0x01EA, 0x01EB, 0x004F],
+    [0x01EC, 0x01ED, 0x004F],
+    [0x01EC, 0x01ED, 0x004F],
+    [0x01EE, 0x01EF, 0x01B7],
+    [0x01EE, 0x01EF, 0x01B7],
+    [0x01F0, 0x01F0, 0x004A],
+    [0x01F1, 0x01F3, 0x01F1],
+    [0x01F1, 0x01F3, 0x01F1],
+    [0x01F1, 0x01F3, 0x01F1],
+    [0x01F4, 0x01F5, 0x0047],
+    [0x01F4, 0x01F5, 0x0047],
+    [0x01F6, 0x0195, 0x01F6],
+    [0x01F7, 0x01BF, 0x01F7],
+    [0x01F8, 0x01F9, 0x004E],
+    [0x01F8, 0x01F9, 0x004E],
+    [0x01FA, 0x01FB, 0x0041],
+    [0x01FA, 0x01FB, 0x0041],
+    [0x01FC, 0x01FD, 0x00C6],
+    [0x01FC, 0x01FD, 0x00C6],
+    [0x01FE, 0x01FF, 0x00D8],
+    [0x01FE, 0x01FF, 0x00D8],
+];
+
+pub struct CollateUtf8Mb4GeneralCi;
```

Comment:
How about `Utf8Mb4GeneralCiCollator`? Also change the trait `collation` to `collator`.